<!-- move base -->
<launch>
  <master auto="start"/>

  <arg name="namespace"/>
  <arg name="scan_topic"  default="scan" />
  <arg name="base_frame"  default="base_footprint"/>
  <arg name="odom_frame"  default="odom"/>

<!--
<param name="use_sim_time" value="true" />
-->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
    <param name="map_frame" value="$(arg namespace)/map"/>
    <param name="base_frame" value="$(arg namespace)/base_link"/>
    <param name="odom_frame" value="/odom"/>
    <param name="map_update_interval" value="0.01"/>
    <param name="maxUrange" value="8.0"/>
    <param name="maxRange" value="10.0"/>	<!-- maxUrange<=maxRange -->
    <param name="sigma" value="0.05"/>
    <param name="kernelSize" value="3"/>
    <param name="lstep" value="0.05"/>
    <param name="astep" value="0.05"/>
    <param name="iterations" value="5"/>
    <param name="lsigma" value="0.075"/>
    <param name="ogain" value="3.0"/>
    <param name="lskip" value="0"/>
    <param name="minimumScore" value="30"/>	<!--scan match score-->
    <param name="srr" value="0.01"/>
    <param name="srt" value="0.02"/>
    <param name="str" value="0.01"/>
    <param name="stt" value="0.02"/>
    <param name="linearUpdate" value="0.05"/>
    <param name="angularUpdate" value="0.0436"/>
    <param name="temporalUpdate" value="-1.0"/>
    <param name="resampleThreshold" value="0.5"/>
    <param name="particles" value="8"/>
  <!--
    <param name="xmin" value="-50.0"/>
    <param name="ymin" value="-50.0"/>
    <param name="xmax" value="50.0"/>
    <param name="ymax" value="50.0"/>
  make the starting size small for the benefit of the Android client's memory...
  -->
    <param name="xmin" value="-1.0"/>
    <param name="ymin" value="-1.0"/>
    <param name="xmax" value="1.0"/>
    <param name="ymax" value="1.0"/>

    <param name="delta" value="0.05"/>		<!--resolution-->
    <param name="llsamplerange" value="0.01"/>
    <param name="llsamplestep" value="0.01"/>
    <param name="lasamplerange" value="0.005"/>
    <param name="lasamplestep" value="0.005"/>
    <remap from="scan" to="$(arg scan_topic)"/>
  </node>


   <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen" clear_params="true">
    <param name="footprint_padding" value="0.01" />
   
    <param name="oscillation_timeout" value="30.0" />
    <param name="oscillation_distance" value="0.5" />
  
    
    <remap from="cmd_vel" to="cmd_vel"/>

    <param name="recovery_behavior_enabled" value="false" />
    <rosparam file="$(find hrobot_nav)/config/odom/ding/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find hrobot_nav)/config/odom/ding/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find hrobot_nav)/config/odom/ding/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find hrobot_nav)/config/odom/ding/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find hrobot_nav)/config/odom/ding/base_global_planner_param.yaml" command="load" />

    <rosparam file="$(find hrobot_nav)/config/odom/ding/teb_local_planner_params.yaml" command="load" />

    <param name="base_global_planner" value="global_planner/GlobalPlanner"/> 
    <param name="planner_frequency" value="1.0" />
    <param name="planner_patience" value="5.0" />

    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="controller_frequency" value="5.0" />
    <param name="controller_patience" value="15.0" />

    <param name="global_costmap/global_frame" value="$(arg namespace)/map"/>
    <param name="global_costmap/robot_base_frame" value="$(arg namespace)/base_link"/>
    <param name="global_costmap/laser_scan_sensor/sensor_frame" value="/$(arg namespace)/laser"/>
    <param name="global_costmap/laser_scan_sensor/topic" value="/$(arg namespace)/scan"/>    
    <param name="local_costmap/global_frame" value="/odom"/>
    <param name="local_costmap/robot_base_frame" value="$(arg namespace)/base_link"/>
    <param name="local_costmap/laser_scan_sensor/sensor_frame" value="$(arg namespace)/laser"/>
    <param name="local_costmap/laser_scan_sensor/topic" value="/$(arg namespace)/scan"/>
    <param name="local_costmap/obstacle_layer/laser_scan_sensor/topic" value="/$(arg namespace)/scan"/>

    

  </node>
</launch>
